router/pkg/config/config.schema.json

```diff
@@ -1716,9 +1716,125 @@
+                  "default": false
+                }
+              }
+            },
+            "root_fields": {
+              "type": "object",
+              "description": "How many root fields are allowed in a query.",
+              "additionalProperties": false,
+              "properties": {
+                "enabled": {
+                  "type": "boolean",
+                  "default": false,
+                  "description": "Enable the root field query limit."
+                },
+                "limit": {
+                  "type": "integer",
+                  "description": "The root field limit for query. If the limit is 0, this limit isn't applied.",
+                  "default": 0,
+                  "minimum": 0
+                },
+                "ignore_persisted_operations": {
+                  "type": "boolean",
+                  "description": "Disable the root fieldlimit for persisted operations. Since persisted operations are stored intentionally, users may want to disable the limit to consciously allow nested persisted operations.",
+                  "default": false
+                }
+              }
+            },
+            "root_field_aliases": {
+              "type": "object",
+              "description": "How many root field aliases are allowed in a query.",
+              "additionalProperties": false,
+              "properties": {
+                "enabled": {
+                  "type": "boolean",
+                  "default": false,
+                  "description": "Enable the root field alias query limit."
+                },
+                "limit": {
+                  "type": "integer",
+                  "description": "The root field alias limit for query. If the limit is 0, this limit isn't applied.",
+                  "default": 0,
+                  "minimum": 0
+                },
+                "ignore_persisted_operations": {
+                  "type": "boolean",
+                  "description": "Disable the root field alias limit for persisted operations. Since persisted operations are stored intentionally, users may want to disable the limit to consciously allow nested persisted operations.",
+                  "default": false
+                }
+              }
+            }
+          }
+        },
         "depth_limit": {
           "type": "object",
-          "description": "The configuration for adding a max depth limit for query (how many nested levels you can have in a query). This limit prevents infinite querying, and also limits the size of the data returned. If the limit is 0, this limit isn't applied.",
+          "description": "DEPRECATED (move to complexity_limits.depth): The configuration for adding a max depth limit for query (how many nested levels you can have in a query).",
```

Comment:
Have you considered following this spec?
https://www.learnjsonschema.com/2020-12/meta-data/deprecated/